testsuites/syncgateway/functional/tests/test_webhooks.py

```diff
@@ -310,6 +314,237 @@ def test_webhooks_crud(params_from_base_test_setup, sg_conf_name, filtered):
+
+    def update_non_webhook_prop():
+        return {'updates': 0, 'data': 'non_webhook_filter'}
+    log_info('Adding {} docs via SDK ...')
+    sdk_doc_bodies = document.create_docs(sdk_webhook, number=sdk_webhook_docs, content={"data": "webhook_filter"}, channels=channel, prop_generator=update_webhook_prop)
+    sdk_doc_ids1 = [doc['_id'] for doc in sdk_doc_bodies]
+    sdk_docs = {doc['_id']: doc for doc in sdk_doc_bodies}
+    sdk_client.upsert_multi(sdk_docs)
+
+    sdk_doc_bodies = document.create_docs(sdk_non_webhook, number=sdk_non_webhook_docs, content={"data": "non_webhook_filter"}, channels=channel)
+    sdk_docs = {doc['_id']: doc for doc in sdk_doc_bodies}
+    sdk_client.upsert_multi(sdk_docs)
+
+    count = 0
+    retries = 5
+    while count < retries:
+        data = webhook_server.get_data()
+        # Remove unwanted data from the response
+        for item in data:
+            if "_id" not in item:
+                data.remove(item)
+
+        posted_webhook_events_ids = [item['_id'] for item in data]
+        if len(posted_webhook_events_ids) < len(sdk_doc_ids1):
+            time.sleep(2)
+            count += 1
+            continue
+        else:
+            break
+    assert len(posted_webhook_events_ids) == len(sdk_doc_ids1)
+
+
+@pytest.mark.syncgateway
+@pytest.mark.webhooks
+def test_webhook_filter_external_https_js(params_from_base_test_setup, setup_webserver_jssslon):
+    """
+    "1. Create valid js function  for import filter and host it in local machine to access jscode with https url
+    webhook filter : add doc.type if doc has data with webhook filter value, otherwise add doc.type is ignore
+    2. Create sgw config and point the webhook filter jsfunction to the webhook function hosted in the localmachine and can be accessed by https url
+    Start sync gateway
+    3. Verify SGW starts sucessfully
+    4. create docs in SDK few docs with data as webhook_filter and some without non_webhook_filter data
+    5. Verfiy webhook events generated and the docs with webhook filter should have in in the events
+    """
+
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    mode = params_from_base_test_setup["mode"]
+    sync_gateway_version = params_from_base_test_setup["sync_gateway_version"]
+    ssl_enabled = params_from_base_test_setup["ssl_enabled"]
+    webhook_server = setup_webserver_jssslon["webhook_server"]
+    sg_conf_name = "webhooks/webhook_filter_external_js"
+
+    if sync_gateway_version < "3.0.0":
+        pytest.skip("this feature not available below 3.0.0")
+    sg_conf = sync_gateway_config_path_for_mode(sg_conf_name, mode)
+    cluster_helper = ClusterKeywords(cluster_config)
+    cluster_hosts = cluster_helper.get_cluster_topology(cluster_config)
+    sg_admin_url = cluster_hosts["sync_gateways"][0]["admin"]
+    sg_url = cluster_hosts["sync_gateways"][0]["public"]
+    username = "autotest"
+    password = "password"
```

Comment:
It seems like this test do not required reset sg, can we avoid to save some execution time . (Change the username & password  and have the unique session this test , when you decided to skip)